src/app.rs

```diff
@@ -137,13 +136,42 @@ where
         self
     }
 
-    /// Set application level arbitrary data item.
+    /// Set application-level arbitrary data item.
     ///
     /// Application data stored with `App::app_data()` method is available
     /// via `HttpRequest::app_data()` method at runtime.
     ///
-    /// This method could be used for storing `Data<T>` as well, in that case
-    /// data could be accessed by using `Data<T>` extractor.
+    /// If this method is used to store a `Data<T>`, that
+    /// data could be accessed by using the `Data<T>` extractor.
+    ///
+    /// # Examples
+    ///
+    /// This is an example to have shared global state
+    /// that would otherwise go inside an `Arc`.
```

Comment:
But it does end up in an Arc inside of Data. Feels like this example is the same as the example on `App::data`. What about an example of something that either wouldn't work or is more efficient using `.data`.